ports/servoshell/egl/ohos.rs

```diff
@@ -0,0 +1,626 @@
+                            (touch_point.x, touch_point.y)
+                        } else {
+                            error!("Move Event received, but no previous touch event was stored!");
+                            // todo: handle this error case
+                            panic!("Move Event received, but no previous touch event was stored!");
+                        }
+                    };
+                    let dx = touch_event.x - lastX;
+                    let dy = touch_event.y - lastY;
+                    TouchEventType::Scroll { dx, dy }
+                } else {
+                    TouchEventType::Move
+                }
+            },
+            OH_NativeXComponent_TouchEventType::OH_NATIVEXCOMPONENT_CANCEL => {
+                if touch_event.id == 0 {
+                    unsafe {
+                        let old = TS_THREAD_STATE.velocity_tracker.take();
+                        assert!(old.is_some());
+                    }
+                }
+                TouchEventType::Cancel
+            },
+            _ => {
+                error!(
+                    "Failed to dispatch call for touch Event {:?}",
+                    touch_event.type_
+                );
+                TouchEventType::Unknown
+            },
+        };
+    if let Err(e) = call(ServoAction::TouchEvent {
+        kind,
+        x: touch_event.x,
+        y: touch_event.y,
+        pointer_id: touch_event.id,
+    }) {
+        error!("Failed to dispatch call for touch Event {kind:?}: {e:?}");
+    }
+}
+
+fn initialize_logging_once() {
+    static ONCE: Once = Once::new();
+    ONCE.call_once(|| {
+        let mut builder = hilog::Builder::new();
+        builder.filter_level(LevelFilter::Debug).init();
+
+        info!("Servo Register callback called!");
+
+        std::panic::set_hook(Box::new(|info| {
+            error!("Panic in Rust code");
+            error!("PanicInfo: {info}");
+            let msg = match info.payload().downcast_ref::<&'static str>() {
+                Some(s) => *s,
+                None => match info.payload().downcast_ref::<String>() {
+                    Some(s) => &**s,
+                    None => "Box<Any>",
+                },
+            };
+            let current_thread = thread::current();
+            let name = current_thread.name().unwrap_or("<unnamed>");
+            let stderr = std::io::stderr();
+            if let Some(location) = info.location() {
+                let _ = error!(
+                    "{} (thread {}, at {}:{})",
+                    msg,
+                    name,
+                    location.file(),
+                    location.line()
+                );
+            } else {
+                let _ = error!("{} (thread {})", msg, name);
+            }
+
+            let _ = crate::backtrace::print_ohos();
+
+            // if opts::get().hard_fail && !opts::get().multiprocess {
```

Comment:
This is the other comment block I was wondering about.  It could have a `TODO:` comment attached if we plan to enable it later.

Reply:
I've removed the commented code for now. I think this panic handler is anyway a likely candidate to be merged with the corresponding desktop implementation, so we can revisit the hard exit option at that point.